crypto/s2n_hkdf.c

```diff
@@ -126,14 +126,22 @@ static int s2n_libcrypto_hkdf_extract(struct s2n_hmac_state *hmac, s2n_hmac_algo
     const EVP_MD *digest = NULL;
     POSIX_GUARD_RESULT(s2n_hmac_md_from_alg(alg, &digest));
 
-    size_t pseudo_rand_key_len = (size_t) pseudo_rand_key->size;
-    POSIX_GUARD_OSSL(HKDF_extract(pseudo_rand_key->data, &pseudo_rand_key_len, digest, key->data, key->size,
+    /* The out_len argument of HKDF_extract is set to the number of bytes written to out_key, and
+     * is not used to ensure that out_key is large enough to contain the PRK. Ensure that the PRK
+     * output will fit in the blob.
+     */
+    uint8_t hmac_size = 0;
+    POSIX_GUARD(s2n_hmac_digest_size(alg, &hmac_size));
+    POSIX_ENSURE(hmac_size <= pseudo_rand_key->size, S2N_ERR_HKDF_OUTPUT_SIZE);
+
+    size_t bytes_written = 0;
```

Comment:
So instead of making "out_len" do double-duty as both out_size and bytes_written, they just don't take an out_size at all and assume?! D: I don't like the "double-duty" model, but no size at all for a `uint8_t*` seems much worse!